### PR TITLE
fix(accordion)!: heading tags

### LINKED ISSUE #1637: [bug] Incorrect text styling in `<rh-accordion-header> `
### Describe the bug

It looks like the text in the accordion header is pulling font styles for buttons. 

Screenshot 2024-06-18 at 9 21 14 AM

### Which Component?

rh-accordion

### Steps to reproduce

View an accordion on ux.redhat.com, like the ones on the [accordion's Code subpage](https://ux.redhat.com/elements/accordion/code/).

### Expected behaviour

The font size should be 18px (`--rh-font-size-body-text-lg`). Weight should be medium (`--rh-font-weight-body-text-medium`).

### Screenshots

_No response_

### Element HTML

_No response_

### Operating System (OS)

macOS

### Browser

_No response_

### More browser information

_No response_

### Additional Info

_No response_

### LINKED ISSUE #1675: [bug] `<rh-accordion-header>` using `heading-tag` attribute breaks header layout
### Describe the bug

When using `rh-accordion-header` the arrow is supposed to be at the far right of the container. However, if the `heading-tag` attribute is used, this breaks. 

This is what the heading should look like:
Screenshot 2024-06-26 at 14-46-38 https __prod foo redhat com 1337

This is what the heading looks like when the attribute is used:
Screenshot 2024-06-26 at 14-46-27 https __prod foo redhat com 1337

The code being used:
```
 <rh-accordion-header heading-tag="h2" expanded="true">
      <h2>Products</h2>
 </rh-accordion-header>
```

The code as it shows up in the inspector:
image

### Screenshots

_No response_

### Steps to reproduce

1. Go to the rh-accordion demo
2. Inspect one of the demo accordions
3. Go to the `rh-accordion-header` inside one
4. Add `heading-tag="h2"` as an attribute
5. The arrow will move from the far right of the container to right next to the text

### Expected behaviour

The arrow shouldn't move

### Element HTML

_No response_

### Operating System (OS)

macOS

### Browser

_No response_

### Additional browser or assistive technology info

_No response_

### PR DESCRIPTION
**Pull request by bennypowers**
also improves perf

Closes #1675 
Closes #1637 

## What I did

1. remove most of the js in rh-accordion-header
2. remove the anonymous slot from rh-accordion-header
3. calculate the shadow header based on child content
4. handle incorrect values for `heading-tag` attr (e.g. `2` instead of `h2`)

## Testing Instructions

1. carefully read the unit tests, ensure that they accurately express user expectations

### CODE REVIEW

**Comment by markcaron** [comment]
@bennypowers there are a few `<rh-accordion-header>` bugs that may or may not be related (perhaps indirectly) to some of the work you're doing here.

Can we tackle one or more of these?

1. #1637 
2. #1675 

And, then there's also #1609

File: elements/rh-accordion/rh-accordion-header.ts [hunk 1]
```diff
@@ -7,9 +7,10 @@ import { customElement } from 'lit/decorators/custom-element.js';
 import { property } from 'lit/decorators/property.js';
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-accordion/rh-accordion-header.ts:7]: Side note: the line numbers referenced here and below are not relevant to the comments.

...Probably should have just added these three points as a comment. /shrug.

[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-accordion/rh-accordion-header.ts:7]: that was the idea, no?
if js fails, fall back to the heading elements,
if js loads, use the content of the headings but remove the element

are you saying we need to in that case add aria heading level to the heading elements afterwards?

[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-accordion/rh-accordion-header.ts:7]: The appropriate approach would be:

```html
<h3>
  <button>Attributes omitted for brevity</button>
</h3>
```

Detecting which heading level to use based off of content from lightdom / users.

Or perhaps using [ARIA heading roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role).

Is it feasible to make the `#container` a heading element (or use ARIA heading roles there)?

[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-accordion/rh-accordion-header.ts:7]: If we apply aria level to the ElementInternals of the rh-accordion-header element, that ought to do i think

[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-accordion/rh-accordion-header.ts:7]: Yes, and `role="heading"`.

**Review by marionnegp — CHANGES_REQUESTED** [review]
I'm still seeing the wrong text style in this branch's demos when I open them in a separate tab. The font size should be 18px (`--rh-font-size-body-text-lg`). Weight should be medium (`--rh-font-weight-body-text-medium`).

I'm also seeing a space below the heading text, which may be because the text isn't large enough to fill that space. Here's what spacing looks like with the caret icon using the updated Brand icon
Screenshot 2024-08-20 at 3 19 02 PM
 in Figma:

File: elements/rh-accordion/rh-accordion-header.css [hunk 2]
```diff
@@ -80,8 +80,8 @@ a {
     var(--_padding-block-end)
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply elements/rh-accordion/rh-accordion-header.css:80]:
```
.toggle { width: 100% }
```
is the issue.

File: elements/rh-footer/rh-footer-lightdom.css [hunk 3]
```diff
@@ -31,7 +31,7 @@
   display: block;
 }
 
-:is(rh-footer) a[slot^="logo"]>img {
+:is(rh-footer) a[slot^="logo"] > img {
```
Comment (adamjohnson) [inline elements/rh-footer/rh-footer-lightdom.css:34]: I submitted a fix for this linting error upstream: https://github.com/RedHat-UX/red-hat-design-system/pull/1777

File: elements/rh-accordion/rh-accordion-header.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-accordion/rh-accordion-header.ts:7]: if that's the case then it seems we should restore the button element in the shadow root

File: elements/rh-accordion/rh-accordion-header.css [hunk 4]
```diff
@@ -80,8 +80,8 @@ a {
...
     var(--_padding-inline-start);
   font-family: var(--rh-font-family-body-text, RedHatText, "Red Hat Text", "Noto Sans Arabic", "Noto Sans Hebrew", "Noto Sans JP", "Noto Sans KR", "Noto Sans Malayalam", "Noto Sans SC", "Noto Sans TC", "Noto Sans Thai", Helvetica, Arial, sans-serif);
-  font-size: var(--_font-size, var(--rh-font-size-body-text-md, 1rem));
-  font-weight: var(--rh-font-weight-heading-medium, 500);
+  font-size: var(--_font-size, var(--rh-font-size-body-text-lg, 1.125rem));
+  font-weight: var(--rh-font-weight-body-text-medium, 500);
   color: var(--_text-color);
 }
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-accordion/rh-accordion-header.css:87]: i restored the `<button>` in shadow root, styles for that already existed

File: elements/rh-accordion/rh-accordion-header.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bennypowers) [reply elements/rh-accordion/rh-accordion-header.ts:7]: I restored the shadow `<button>`, as a result, I had to change the type parameter of the tabindex controller to HTMLButtonElement, which required some fancy footwork with `hasUpdated`, yadda yadda. Also had to move some event listeners around to appease the a11y linter, which i think should all be improvements. firefox' audit still complains that there are buttons on the page with [tabindex=-1] but i think that's a false positive, cause, y'know, rti

I added a heading level controller for when the headers are not wrapped

playwright's ax tree doesn't expose aria-level so i had to fall back to `.is.accessible()`

**Comment by bennypowers** [comment]
> I'm still seeing the wrong text style in this branch's demos when I open them in a separate tab. The font size should be 18px (`--rh-font-size-body-text-lg`). Weight should be medium (`--rh-font-weight-body-text-medium`).
> 
> I'm also seeing a space below the heading text, which may be because the text isn't large enough to fill that space. Here's what spacing looks like with the caret icon using the updated Brand icon Screenshot 2024-08-20 at 3 19 02 PM in Figma:

I change things so that the heading font-size is always large. this means that now the `large` boolean only changes the padding of the panel, but not the font size of the header. That contradicts the docs:

> ### Sizes
> There are two available sizes and the only difference is the title text size. You can use the Small size on large breakpoints, but not the Large size on small breakpoints due to the potential of long title text wrapping to more than two lines.

**Pull request merged**